classes/Dispatcher.php

```diff
@@ -1184,4 +1184,52 @@ public static function getControllersInDirectory($dir)
 
         return $controllers;
     }
+
+    /**
+     * Get the default php_self value of a controller.
+     *
+     * @param string $controller The controller class name
+     *
+     * @return string|null
+     */
+    public static function getControllerPhpself($controller)
+    {
+        if (!class_exists($controller)) {
+            return;
+        }
+
+        $reflectionClass = new ReflectionClass($controller);
+        $controllerDefaultProperties = $reflectionClass->getDefaultProperties();
+
+        return isset($controllerDefaultProperties['php_self']) ? $controllerDefaultProperties['php_self'] : null;
+    }
+
+    /**
+     * Get list of all php_self property values of each available controller in the specified dir.
+     *
+     * @param string $dir Directory to scan (recursively)
+     * @param bool $base_name_otherwise Return the controller base name if no php_self is found
+     *
+     * @return array
+     */
+    public static function getControllersPhpself($dirs, $base_name_otherwise = true)
```

Comment:
Could you please give this method a name more distinct that the other one? I understand that you're adding an `s` because one method returns one controller and the other returns more than one, but it's error prone in the end.

Maybe add the word **list**, something like `getControllerPhpselfList` or **many**, like `getManyControllersPhpself`, or something else, it's up to you ;)

Reply:
@matthieu-rolland thank you for your toooooooooooooo... late feedback, i don't understand why this repository is so sleepy, 16 days for such a feedback ? Should i rebase my work and wait like another month to mege it, and may be 5 months to release it ??!! To be honest, i made this PR because i needed it for some project, and projects can't wait all that time to fix such a bug, so don't expect from people to continue contributing if they must wait all that time. (prière de passer ce feedback à votre équipe)

Reply:
Hello @dali-rajab 

While your contribution is very much welcome, passive-aggressive comments are definitely not. Let me remind you that we don't work for you, and kindly ask you to respect our [Code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).

Regarding the so-called "sleepyness", have you ever occurred to you that [you're not the only one opening PRs](https://github.com/pulls?q=is%3Apr+archived%3Afalse+org%3APrestaShop+created%3A%3E%3D2020-05-16+sort%3Aupdated-asc+)?

Reply:
@eternoendless @matthieu-rolland thank you for your fast reactions (sorry for bothering you)

Reply:
@eternoendless @matthieu-rolland : I updated the PR here https://github.com/PrestaShop/PrestaShop/pull/21168, so please close this one and reference the new one.

Thank you very much.